### PR TITLE
Support for free orientation

### PR DESCRIPTION
**Pull request by anujanegi**
Things added-
 - Free orientation dummy data generation
 - Testing support for free orientation for gamma map (with dummy data only).

### CODE REVIEW

**Comment by anujanegi** [comment]
Going okay I think. No major obstacle currently!
Currently, running simulations again to generate lead fields for free orientation and expand testing with these "real" leadfileds!

**Comment by agramfort** [comment]
@anujanegi let me know if you want me to review

**Comment by agramfort** [comment]
@anujanegi can you update me on your status? what's working and what's not? what are the next TODOs? 🙏

**Comment by anujanegi** [comment]
### What's working/what is added:

 - brain plots viz (for nnz = 1, 2, 3) 
 - Added performance metrics (hamming, jaccard, recall)

Support for free orientation (working and tested)[TBD]:
 - Estimators (nnz>1, t=1, 10, fixed/free orient) 
 - Testing (nnz>1, t>=1, 10, fixed/free orient) 
 - Dummy data generation for free orient 
 - Better parameterization 
 - Viz(nnz>1, t=10, fixed)[viz for free orient]  

### Current issues/TODOs:
 - Running simulation for generating free orientation leadfields and then adding them in the testing and viz pipeline.
 - Euclidean distance check for nnz>1

**Comment by agramfort** [comment]
cool ! do you think it would be possible to share some figures or a few slides so demo the progress? 🙏

**Comment by anujanegi** [comment]
Visualisations for fixed orientation cases :

| **Estimator/nnz**    | 1 | 2 | 3 |
|----------------------|---|---|---|
| Gamma Map     | gamma_map_CC120264  | gamma_map_CC120264  |  gamma_map_CC120264|
| Iterative L1     | iterative_L1_CC120264  |  iterative_L1_CC120264 |  iterative_L1_CC120264 |
| Iterative L1 Type II | iterative_L1_typeII_CC120264 | iterative_L1_typeII_CC120264 |  iterative_L1_typeII_CC120264 |
| Iterative L2         |  iterative_L2_CC120264 | iterative_L2_CC120264  |  iterative_L2_CC120264 |
| Iterative L2 Type II |  iterative_L2_typeII_CC120264 | iterative_L2_typeII_CC120264 |  iterative_L2_typeII_CC120264 |
| Iterative Sqrt     |iterative_sqrt_CC120264 |  iterative_sqrt_CC120264|  iterative_sqrt_CC120264 |


*P.S. given viz is for subject 20264*

**Comment by agramfort** [comment]
@anujanegi can you summarize in a few sentences what you observe from these plots? is it working as you have expected? anything suspicious? 🙏
